README.md

```diff
@@ -93,20 +150,79 @@ Some trust is required in the PLC server. Its attacks are limited to:
 - Misordering: In the event of a fork in DID document history, the server could choose to serve the "wrong" fork
 
 
+### DID Creation
+
+To summarize the process of creating a new `did:plc` identifier:
+
+- collect values for all of the core data fields, including generating new secure key pairs if necessary
+- construct an "unsigned" regular operation object. include a `prev` field with `null` value. do not use the deprecated/legacy operation format for new DID creations
+- serialize the "unsigned" operation with DAG-CBOR, and sign the resulting bytes with one of the initial `rotationKeys`. encode the signature as `base64url`, and use that to construct a "signed" operation object
+- serialize the "signed" operation with DAG-CBOR, take the SHA-256 hash of those bytes, and encode the hash bytes in `base32`. use the first 24 characters to generate DID value (`did:plc:<hashchars>`)
+- serialize the "signed" operation as simple JSON, and submit it via HTTP POST to `https://plc.directory/:did`
+- if the HTTP status code is successful, the DID has been registered
+
+When "signing" using a "`rotationKey`", what is meant is to sign using the private key associated the public key in the `rotationKey` list.
+
+### DID Update
+
+To summarize the process of updating a new `did:plc` identifier:
+
+- if the current DID state isn't known, fetch the current state from `https://plc.directory/:did/data`
+- if the most recent valid DID operation CID (hash) isn't known, fetch the audit log from `https://plc.directory/:did/log/audit`, identify the most recent valid operation, and get the `cid` value. if this is a recovery operation, the relevant "valid" operation to fork from may not be the most recent in the audit log
+- collect updated values for all of the core data fields, including generating new secure key pairs if necessary (eg, key rotation)
+- construct an "unsigned" regular operation object. include a `prev` field with the CID (hash) of the previous valid operation
+- serialize the "unsigned" operation with DAG-CBOR, and sign the resulting bytes with one of the previously-existing `rotationKeys`. encode the signature as `base64url`, and use that to construct a "signed" operation object
+- serialize the "signed" operation as simple JSON, and submit it via HTTP POST to `https://plc.directory/:did`
+- if the HTTP status code is successful, the DID has been updated
+- the DID update may be nullified by a "rotation" operation during the recovery window (currently 72hr)
+
+### DID Deactivation
+
+To summarize the process of de-activating an existing `did:plc` identifier:
+
+- if the most recent valid DID operation CID (hash) isn't known, fetch the audit log from `https://plc.directory/:did/log/audit`, identify the most recent valid operation, and get the `cid` value
```

Comment:
this should actually be done at `/log` which does not include invalidated ops

Reply:
(applies to above paragraph as well)

Reply:
You do actually need the full `/log/audit` to get the `cid` field (along with `createdAt`, etc). We could change `/log` to include that extra metadata as well? Right now it is just an array of the ops, no extra server-side metadata